wpe/src/main/cpp/Browser/Page.h

```diff
@@ -23,76 +23,45 @@
 #pragma once
 
 #include "InputMethodContext.h"
-#include "PageEventObserver.h"
-#include "PageSettings.h"
+#include "JNI/JNI.h"
 #include "Renderer.h"
 
-#include <memory>
-#include <string>
 #include <vector>
-#include <wpe/webkit.h>
-#include <wpe/wpe.h>
-
 #include <wpe-android/view-backend-exportable.h>
 
-struct ANativeWindow;
-struct ExportedBuffer;
+DECLARE_JNI_CLASS_SIGNATURE(JNIPage, "com/wpe/wpe/Page");
 
-class Page final {
+class Page final : public InputMethodContextObserver {
 public:
-    Page(int width, int height, std::shared_ptr<PageEventObserver> observer);
+    static void configureJNIMappings();
 
     Page(Page&&) = delete;
     Page& operator=(Page&&) = delete;
     Page(const Page&) = delete;
     Page& operator=(const Page&) = delete;
 
-    ~Page();
-
-    void init();
-    void close();
-
-    void loadUrl(const char* url);
-    void goBack();
-    void goForward();
-    void stopLoading();
-    void reload();
-
-    void surfaceCreated(ANativeWindow* window);
-    void surfaceChanged(int format, int width, int height);
-    void surfaceRedrawNeeded();
-    void surfaceDestroyed();
-    void handleExportedBuffer(const std::shared_ptr<ExportedBuffer>& exportedBuffer);
-
-    void onTouch(wpe_input_touch_event_raw* touchEventRaw);
-    void setZoomLevel(double zoomLevel);
-
-    void setInputMethodContent(const char c);
-    void deleteInputMethodContent(int offset);
+    ~Page() override { close(); }
 
-    void domFullscreenRequest(bool fullscreen);
-    void requestExitFullscreen();
-    void fullscreenImageReady();
+    void close() noexcept;
 
-    struct wpe_android_view_backend_exportable* exportable() { return m_viewBackendExportable; }
+    WebKitWebView* getWebView() const noexcept { return m_webView; }
```

Comment:
"get" in getter name is wrong according to webkit style guide

Reply:
done